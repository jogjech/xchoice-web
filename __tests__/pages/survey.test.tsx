import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import ViewSurveyPage from "../../pages/survey";

const surveyId = "surveyId";

jest.mock("next/router", () => ({
  useRouter: () => {
    return {
      route: "/",
      pathname: "",
      query: { id: surveyId },
    };
  },
}));

describe("View survey page", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<ViewSurveyPage></ViewSurveyPage>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains layout", () => {
    const layout = wrapper.find("Layout");
    expect(layout.exists()).toBeTruthy();
    expect(layout.prop("title")).toBe("View Survey");
  });

  it("contains survey viewer", () => {
    const surveyViewer = wrapper.find("SurveyViewer");
    expect(surveyViewer.exists()).toBeTruthy();
    expect(surveyViewer.prop("surveyId")).toBe(surveyId);
  });
});
