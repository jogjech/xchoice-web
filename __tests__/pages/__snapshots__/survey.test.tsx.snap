// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View survey page renders correctly 1`] = `
<Layout
  title="View Survey"
>
  <Memo(BackTop) />
  <SurveyViewer
    surveyId="surveyId"
  />
</Layout>
`;
